### ISSUE TITLE
YOL) compiles and runs, but no predictions

### ISSUE THREAD

**Issue by ramanpreet9** [pjreddie/darknet#486]
I compiled yolo with GPU, CUDA, and opencv on.
It compiles correctly. it runs on the sample images like dog.jpg etc as described in wiki. However, I am not getting any predictions!

My output:
Case 1 and 2 (GPU ON, CUDA ON/OFF, opencv on
Loading weights from yolo.weights...Done!
data/dog.jpg: Predicted in 0.056 seconds.

and it hangs here

case 3: GPU ON,              CUDA, OPENCV  OFF
data/dog.jpg: Predicted in 0.17 seconds.
Not compiled with OpenCV, saving to predictions.png instead

but the predictions have no BB.

One thing i noticed was that the link for yolo weights is ~200MB but the wiki page says its 1 GB.

**Comment by konbick** [comment]
I am having exactly the same problem and also the same observation about the file size.

Were you able to solve this problem meanwhile?

**Comment by ramanpreet9** [comment]
The yolo wights have an issue as i mentioned in my post. If you use the small or tiny weights, it works fine. but, until the author responds, you cannot use the full weights.

**Comment by konbick** [comment]
Man, thanks a lot for your post! You saved me a lot of headache!

**Comment by konbick** [comment]
Oh, no. It doesn't work for me even with the tiny weights...

**Comment by ramanpreet9** [comment]
try both tiny and small. it worked for me.

**Comment by konbick** [comment]
Should I use a different cfg? Could you let me know your command when trying to infere a test image? Thanks

**Comment by ramanpreet9** [comment]
./darknet yolo test cfg/yolov1/yolo-small.cfg /weights/yolo-small.weights
data/dog.jpg -thresh 0.1

**Comment by ramanpreet9** [comment]
Yes, you need to use the relevant config file   small.weights with small.cfg  etc

**Comment by konbick** [comment]
If I use the following command, it works well for me, too

./darknet detect cfg/yolo.cfg weights/yolo.weights data/horses.jpg

**Comment by csbrown** [comment]
I am having the same problem, except I cannot detect the dog under for any combination of weights and architectures.

CUDA 8.0
K80
CUDNN 7.1
Nvidia drivers 375.66

**Comment by csbrown** [comment]
Upgraded my nvidia drivers and cuda.  No better.

CUDA 9.1
K80
CUDNN 7.1
Nvidia drivers 390.67

GPU=1, CUDNN=1

**Comment by csbrown** [comment]
Lowered the `-thresh` option to 0.01, and it detects everything, everywhere at a confidence of 5%.  This is weird, but I suppose this is because 0%s cause issues with cross-entropy loss, so the minimum confidence is 5%..... okay.  If I set the `-thresh` option to 0.06, I detect like a gazillion people, one for each grid cell, it seems.  Going to add a cli option for -hier_thresh and see what that does so I can get a handle on all this to dig deeper to debug.

**Comment by csbrown** [comment]
Turned off CUDNN and it seems to work.  What version of CUDNN is everyone using?

**Comment by grosmar** [comment]
Any update?

I have the same issue with Cuda 10, if I disable CUDNN it works

**Comment by Tzuya14** [comment]
I encountered the same problem while working on GCP and I can confirm that it works every time while disabling cudnn but I also made it work with cudnn enabled just by creating a new IDENTICAL virtual machine with same cuda and cudnn, which is a bit weird. This may show that is not a problem with the weights.

**Comment by ministrudels** [comment]
I encountered this problem using the NVIDIA GPU Cloud Image for Deep Learning, Data Science, and HPC image on GCP. Turns out, it was because I was using the  NVIDIA Tesla V100 GPU. I wasnt having this issue on a new instance with the  NVIDIA Tesla P100.

**Comment by SirojbekSafarov** [comment]
I had the same problem. Can anyone solve this problem

**Comment by SirojbekSafarov** [comment]
opencv=1
cuda=1
cudnn=1

same error, please modify cfg/yolov3.cfg,

# Testing
batch=1
subdivisions=1
# Training
#batch=256
#subdivisions=64
then, detector result display ok

**Comment by CheRaton** [comment]
i have the same issue ... 

some update with this problem?

`Performing object detection:
INPUT_IMGS----- tensor([[[[0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          ...,
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.]],

         [[0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          ...,
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.]],

         [[0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          ...,
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.]]],


        [[[0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          ...,
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.]],

         [[0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          ...,
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.]],

         [[0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          ...,
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.]]],


        [[[0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          ...,
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.]],

         [[0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          ...,
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.]],

         [[0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          ...,
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.]]],


        [[[0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          ...,
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.]],

         [[0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          ...,
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.]],

         [[0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          ...,
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.],
          [0., 0., 0.,  ..., 0., 0., 0.]]]])
	+ Batch 0, Inference Time: 0:00:00.487818

Saving images:
(0) Image: 'data/test/DZUY1ekX4AEiNgq.jpg'
(1) Image: 'data/test/Metallic-Corrosion.jpg'
(2) Image: 'data/test/corrosion002.jpg'
(3) Image: 'data/test/istockphoto-1179527589-1024x1024.jpg'
`

**Comment by AbhiKavathiya** [comment]
I have same issue and I have Cuda - v11.7 and Cudnn-11 so give suggestions
